Add 2.18 changelog to docs (#38052)
DataDog/documentation: content/en/observability_pipelines/guide/upgrade_worker.md

This guide goes over how to upgrade to a specific Worker version and the updates for that version.

## Worker version 2.18.0

To upgrade to Worker version 2.18.0:

- Docker: Run the `docker pull` command for the [2.18.0 image][48].
- Kubernetes: See the [Helm chart][2] and [Upgrade the Worker][37].
- APT: Run the command `apt-get install observability-pipelines-worker=2.18.0`.
- RPM: Run the command `sudo yum install observability-pipelines-worker-2.18.0`.

Worker version 2.18.0 gives you access to the following:

#### New features

> - A `--log-format` flag has been added to the `run` command, also configurable using the `DD_OP_LOG_FORMAT` environment variable, for selecting the format of the Worker's own stdout/stderr logs. The default `text` preserves the existing human-readable output, and `json` emits structured JSON logs that can be tailed and parsed directly without additional parsing rules.

Reviewer: ```suggestion
- The `run` command now supports a `--log-format` flag, also configurable using the `DD_OP_LOG_FORMAT` environment variable, for selecting the format of the Worker's own `stdout` and `stderr` logs. The default `text` preserves the existing human-readable output, and `json` emits structured JSON logs that can be tailed and parsed directly without additional parsing rules.
```

Reviewer: The New features entries elsewhere on this page (2.17.0, 2.16.0) consistently use active, declarative present tense ("X now supports/accepts Y") — this is also specified in the [style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#active-voice)

> - A `tracking_mode` option has been added to the Tag Cardinality Limit processor. `exact_fingerprint` tracks tag values exactly using 64-bit fingerprints, which uses less memory than the previous method of storing raw values, and `probabilistic` uses a bloom filter for even lower memory usage in exchange for an occasional false positive. Probabilistic mode accepts a `false_positive_rate` field, which defaults to `0.001` (0.1%).

Reviewer: Same active-voice suggestion as the flag bullet above:

```suggestion
- The Tag Cardinality Limit processor now supports a `tracking_mode` option. `exact_fingerprint` tracks tag values exactly using 64-bit fingerprints, which uses less memory than the previous method of storing raw values, and `probabilistic` uses a bloom filter for even lower memory usage in exchange for an occasional false positive. Probabilistic mode accepts a `false_positive_rate` field, which defaults to `0.001` (0.1%).
```

- The WebSocket source is available for ingesting logs from a WebSocket endpoint, with support for `none`, `basic`, `bearer`, and `custom` authentication strategies, as well as TLS.

> - Added support for the sample processor in traces pipelines.

Reviewer: This bullet's construction ("Added support for...") breaks parallel structure with its sibling bullets. Suggest matching the "X now supports Y" pattern used elsewhere on the page:

```suggestion
- Trace pipelines now support the sample processor.
```

> - A new `generate_metrics` processor has been added that can route metrics to any supported metrics destination. Datadog recommends using it over the existing `generate_datadog_metrics` processor, although use of the latter is not affected.

Reviewer: Same active-voice suggestion:

```suggestion
- A new `generate_metrics` processor routes metrics to any supported metrics destination. Datadog recommends using it over the existing `generate_datadog_metrics` processor, although use of the latter is not affected.
```

#### Enhancements

- mTLS support has been added to the BYOC Logs destination.
- The `measure_cpu_usage` option is now enabled, so that `pipelines.component_cpu_usage_ns_total` is always emitted for the following log processors: Custom Processor, Sensitive Data Scanner, Grok Parser, Parse JSON, Parse XML, OCSF Mapper, Enrichment Table, Reduce, Dedupe, Split Array, and Throttle; and the following metrics processors: Aggregate and Tag Cardinality Limit.

> - Pipeline configuration errors now include the ID of the component that caused the failure, making it easier to identify which source, processor, or destination needs to be corrected. For processors made up of multiple blocks, such as the Custom Processor, Generate Metrics, and OCSF Mapper, the error also identifies which individual block failed, including its position and, where the block has one, its name.

Reviewer: ```suggestion
- Pipeline configuration errors now include the ID of the component that caused the failure. This makes it easier to identify which source, processor, or destination needs correcting. For processors made up of multiple blocks, such as the Custom Processor, Generate Metrics, and OCSF Mapper, the error also identifies which individual block failed, including its position and, where the block has one, its name.
```

Reviewer: The trailing "...making it easier to identify..." clause is a mild AI-writing-style marker (afterthought participial phrase).

> - Pipeline metrics are now tagged with the pipeline name using `pipeline_name:<normalized pipeline name>`.

Reviewer: ```suggestion
- Pipeline metrics are now tagged with the pipeline name using `pipeline_name:<NORMALIZED_PIPELINE_NAME>`.
```

#### Fixes

- Fixed an issue that occurred when a log-only pipeline (using legacy search syntax) ran a search query that the new parser could not handle. The parser now emits a warning and does not run the invalid query.

> - Fixed an issue affecting queries that start with an escaped character (for pipelines running new search syntax only). 

Reviewer: ```suggestion
- Fixed an issue affecting queries that start with an escaped character (for pipelines running new search syntax only).
```

> - Internal telemetry (metrics and logs) emitted from background tasks now correctly inherits the owning component's tags (`component_id`, `component_kind`, `component_type`). Previously, several components spawned background tasks without propagating their tags, so some internal events from those tasks were missing their component tags. Affected emissions include the `datadog_archives` source's listing-error, download-error, send-error, event-acknowledged/errored/rejected, and status metrics from its listing coordinator, per-file downloader, and acknowledgement-handler tasks; the `microsoft_sentinel` destination's Azure OAuth token-regeneration task; and the `splunk_tcp` source's server task.

Reviewer: In addition to the suggestions above, the "Affected emissions include..." clause is currently a bit long and difficult to scan. Would it be possible to update this to a bulleted list? For example:

```
Affected emissions include:
- `datadog_archives` source: listing-error, download-error, send-error, event-acknowledged, event-errored, event-rejected, and status metrics from its listing coordinator, per-file downloader, and acknowledgment-handler tasks
- `microsoft_sentinel` destination: the Azure OAuth token-regeneration task
- `splunk_tcp` source: the server task
```

If the specific task-to-metric mapping is known, the first bullet could be further split to add precision — but I'm open to your feedback!

Author: Made that change

> - Internal telemetry (metrics and logs) emitted from background tasks now correctly inherits the owning component's tags (`component_id`, `component_kind`, `component_type`). Previously, several components spawned background tasks without propagating their tags, so some internal events from those tasks were missing their component tags. Affected emissions include the `datadog_archives` source's listing-error, download-error, send-error, event-acknowledged/errored/rejected, and status metrics from its listing coordinator, per-file downloader, and acknowledgement-handler tasks; the `microsoft_sentinel` destination's Azure OAuth token-regeneration task; and the `splunk_tcp` source's server task.

Reviewer: ```suggestion
- Internal telemetry (metrics and logs) emitted from background tasks now correctly inherits the owning component's tags (`component_id`, `component_kind`, `component_type`). Previously, several components spawned background tasks without propagating their tags, so some internal events from those tasks were missing their component tags. Affected emissions include the `datadog_archives` source's listing-error, download-error, send-error, event-acknowledged, event-errored, and event-rejected, and status metrics from its listing coordinator, per-file downloader, and acknowledgment-handler tasks; the `microsoft_sentinel` destination's Azure OAuth token-regeneration task; and the `splunk_tcp` source's server task.
```
